### PR TITLE
List substitutions

### PR DESCRIPTION
**Pull request by phanimarupaka**
@mortent 

This PR lists substitutions along with setters in table format. The setters which are referenced as part of substitutions will also be displayed.

@yuwenma FYI

### CODE REVIEW

File: kyaml/setters2/list.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ type List struct {
 	Name string
 
 	Setters []SetterDefinition
+
+	Substitutions []SubstitutionDefinition
```
Comment (mortent) [inline kyaml/setters2/list.go:22]: This feels like two separate pieces of functionality that maybe should be separate. Also, the implementation for setters and substitutions look very similar. Could we improve this?

Comment (phanimarupaka) [reply kyaml/setters2/list.go:22]: Setters and substitutions are closely coupled as substitution is combination of existing setters. So listing them together would make it easy for users to deduce relation between them.
For implementation, yes they seem similar in terms of parsing openAPI file, but the metadata is different so better to keep them separate. Updated the method signatures for distinction.

File: cmd/config/go.mod [hunk 2]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/stretchr/testify v1.4.0
 	github.com/ugorji/go/codec v0.0.0-20181204163529-d75b2dcb6bc8 // indirect
 	k8s.io/apimachinery v0.17.0
-	sigs.k8s.io/kustomize/kyaml v0.0.0
+	sigs.k8s.io/kustomize/kyaml v0.1.7
```
[the remark this answers is not in the document]
Comment (phanimarupaka) [reply cmd/config/go.mod:16]: For local dev. It is anyways overridden by local but should have a valid version to be referenced from parent modules.

**Pull request merged**